src/apify_client/clients/resource_clients/dataset.py

```diff
@@ -0,0 +1,227 @@
+                To reverse the order, set this parameter to True.
+            clean (bool, optional): If True, returns only non-empty items and skips hidden fields (i.e. fields starting with the # character).
+                The clean parameter is just a shortcut for skip_hidden=True and skip_empty=True parameters.
+                Note that since some objects might be skipped from the output, that the result might contain less items than the limit value.
+            fields (list[str], optional): A list of fields which should be picked from the items,
+                only these fields will remain in the resulting record objects.
+                Note that the fields in the outputted items are sorted the same way as they are specified in the fields parameter.
+                You can use this feature to effectively fix the output format.
+            omit: (list[str], optional): A list of fields which should be omitted from the items.
+            unwind: (str, optional): Name of a field which should be unwound.
+                If the field is an array then every element of the array will become a separate record and merged with parent object.
+                If the unwound field is an object then it is merged with the parent object.
+                If the unwound field is missing or its value is neither an array nor an object and therefore cannot be merged with a parent object,
+                then the item gets preserved as it is. Note that the unwound items ignore the desc parameter.
+            skip_empty (bool, optional): If True, then empty items are skipped from the output.
+                Note that if used, the results might contain less items than the limit value.
+            skip_hidden (bool, optional): If True, then hidden fields are skipped from the output, i.e. fields starting with the # character.
+
+        Returns:
+            The dataset items
+        """
+        request_params = self._params(
+            offset=offset,
+            limit=limit,
+            desc=desc,
+            clean=clean,
+            fields=fields,
+            omit=omit,
+            unwind=unwind,
+            skipEmpty=skip_empty,
+            skipHidden=skip_hidden,
+        )
+
+        response = self.http_client.call(
+            url=self._url('items'),
+            method='GET',
+            params=request_params,
+        )
+
+        data = response.json()
+
+        return {
+            'items': data,
+            'total': int(response.headers['x-apify-pagination-total']),
+            'offset': int(response.headers['x-apify-pagination-offset']),
+            'count': len(data),  # because x-apify-pagination-count returns invalid values when hidden/empty items are skipped
+            'limit': int(response.headers['x-apify-pagination-limit']),  # API returns 999999999999 when no limit is used
+        }
+
+    def download_items(
```

Comment:
I don't know what the convention is in Python, but I kinda dislike functions that return different types based on some configuration parameter. With `getRecord` I used `stream` and `buffer` as options, because the function doesn't really have any other options. But in case of `downloadItems`, I was thinking of actually creating a different function for streaming, like `createItemStream` In Node, you could then 

```js
createItemStream('csv').pipe(fs.createWriteStream('data.csv'))
```

I think we should agree on this, so that the API stays the same across the clients.

Reply:
+1 for Ondra. I think we could actually have internal function `_download_items` which returns different types but then create thin wrapper for stream and buffer versions.

Reply:
Updated this based on our previous discussion with Ondra to add the function `iterate_items`, which iterates over the dataset items, downloading them as necessary.